src/smtencoding/FStar.SMTEncoding.Z3.fs

```diff
@@ -377,7 +377,13 @@ type job<'a> = {
     callback: 'a -> unit
 }
 
-type z3job = job<(z3status * int * z3statistics)>
+type z3result =
+      z3status
+    * int
+    * z3statistics
+    * option<string> // query hash
```

Comment:
Past two/three fields, I tend to like records better (though I agree that adding the extra type is already a plus)

Reply:
I have no opinion about that, happy to change it to records if that's preferred.